### DISCUSSION TITLE
Type concatenation when using variadic generics

### DISCUSSION THREAD

**Discussion by dnaaun** [microsoft/pyright#2319] [Q&A] [answered]
Once again, hats off to Pyright for taking the lead on implementing the draft PEP 646.

I'm trying to make use of the [type concatenation](https://www.python.org/dev/peps/pep-0646/#type-concatenation) feature of the PEP. The example below is straight from the PEP (except for using `Unpack` instead of asterisks), but Pyright is complaining. I'm wondering if I'm missing something.

```python
from typing import Generic, NewType
from typing_extensions import TypeVarTuple, Unpack


Shape = TypeVarTuple("Shape")
Batch = NewType("Batch", int)


class Array(Generic[Unpack[Shape]]):
    ...


def add_batch_axis(x: Array[Unpack[Shape]]) -> Array[Batch, Unpack[Shape]]: # Pyright complains "TypeVarTuple not allowed in this context"
    ...
```

**Comment by erictraut** [comment, marked as answer]
Thanks for kicking the tires on this support. It hasn't gotten much use yet, which is understandable given that the PEP has not yet been ratified. You're correct that this is a bug. The fix will be in the next release.

**Reply by dnaaun** [reply]
Thanks Eric. I've been longing for variadic generics in Python because it makes my life debugging/writing deep learning models 10x easier, so your efforts are very appreciated.
